Apache-Beam-Jobs.md

```diff
@@ -227,107 +223,357 @@ error_pcoll = (
+        'e4', 'o1', ['K', 'L', 'M', 'N'], ['TextInput'])
+
+    self.assert_job_output_is([
+        job_run_result.JobRunResult(stdout='14'),
+    ])
 ```
 
+Note that `self.assert_job_output_is(...)` and `self.assert_job_output_is_empty()` do as advertised -- they run the job to completion and verify the result.
+
+> **IMPORTANT:** Only one `assert_job_output_is` assertion can be performed in a test body. Multiple calls will result in an exception instructing you to split the test apart.
+
+Just because a job passes in unit tests does not guarantee it will pass in production. This is because workers, which execute the pipeline code, are run in a special environment where the code base is configured differently. While Oppia's jobs team works to resolve the differences, be careful about using complex and/or confusing objects. The simpler your job, the greater chance it'll work in production!
+
+## Running Apache Beam Jobs
+
+### Local Development Server
+
+These instructions assume you are running a local development server. If you are a release coordinator running these jobs on the production or testing servers, you should already have been granted the "Release Coordinator" role, so you can skip steps 1-3.
+
+1. Sign in as an administrator ([instructions][3]).
+2. Navigate to **Admin Page > Roles Tab**.
+3. Add the "Release Coordinator" role to the username you are signed in with.
+4. Navigate to http://localhost:8181/release-coordinator, then to the **Beam Jobs tab**.
+5. Search for your job and then click the **Play button**.
+6. Click "Start new job".
+
+![Screen recording showing how to run jobs](https://user-images.githubusercontent.com/5094060/128743997-70cca5f9-0b76-4294-806e-f65f5df5be95.gif)
+
+### Production Server
+
+Before a job or feature that cannot be fully tested locally can be run and deployed in production, it must first be tested on the Oppia backup server. Examples of features that require backup server testing are features that add or modify code that requires some third-party API, like Cloud Tasks, Cloud Storage, or Cloud Translate.
+
+If your job or feature is not essential for the release and has not been fully tested by the release cut, then it is not going into the release. "Fully tested" means:
+- The job or feature should run without failures on the Oppia backup server.
+- The job or feature produces the expected output.
+- The job or feature has the expected outcome (this must be verified by e.g. user-facing changes, or introspection of the datastore, or a validation job, or an output check, etc.).
+- The job or feature should be explicitly approved by server jobs admin (currently @seanlip and @vojtechjelinek).
+
+Also, in case your job is a new migration job (not any already existing migrations), there has to be an audit job accompanying it to verify that the data that you are migrating is valid in the server. **The audit job will have to be "Fully tested" before testing on the migration job can start.**
```

Comment:
```suggestion
Also, in case your job changes data in the datastore, there has to be a validation job accompanying it to verify that the data that you are migrating is valid in the server. **The validation job will have to be "Fully tested" before testing on the migration job can start.**
```

Reply:
Done.